tools/c_ast/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+use clang_ast::Node;
+use harvest_core::{
+    Id, Representation,
+    tools::{MightWriteContext, MightWriteOutcome, RunContext, Tool},
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    fs::File,
+    process::{Command, Stdio},
+};
+use tracing::info;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct QualType {
+    #[serde(rename = "desugaredQualType")]
+    pub desugared_qual_type: Option<String>,
+    #[serde(rename = "qualType")]
+    pub qual_type: String,
+    #[serde(rename = "typeAliasDecId")]
+    pub type_alias_dec_id: Option<clang_ast::Id>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Clang {
+    TranslationUnitDecl,
+    TypedefDecl {
+        loc: Option<clang_ast::SourceLocation>,
+        range: Option<clang_ast::SourceRange>,
+        name: String,
+        #[serde(rename = "type")]
+        qtype: QualType,
+        annotation: Option<String>,
+    },
+    FunctionDecl {
+        loc: Option<clang_ast::SourceLocation>,
+        range: Option<clang_ast::SourceRange>,
+        name: String,
+        #[serde(rename = "storageClass")]
+        storage_class: Option<String>,
+        #[serde(rename = "type")]
+        qtype: QualType,
+        annotation: Option<FunctionAnnotation>,
+    },
+    ParmVarDecl {
+        loc: Option<clang_ast::SourceLocation>,
+        range: Option<clang_ast::SourceRange>,
+        name: Option<String>,
+        #[serde(rename = "type")]
+        qtype: QualType,
+        annotation: Option<String>,
+    },
+    CompoundStmt {
+        loc: Option<clang_ast::SourceLocation>,
+        range: Option<clang_ast::SourceRange>,
+        annotation: Option<String>,
+    },
+    Other {
+        kind: Option<String>,
+        annotation: Option<String>,
+    },
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum FunctionAnnotation {
+    Entry,
+    Static,
+}
+
+fn annotate_ast(ast: &mut Node<Clang>) {
+    if let Clang::FunctionDecl {
+        name,
+        storage_class,
+        annotation,
+        ..
+    } = &mut ast.kind
+    {
+        match storage_class.as_ref().map(|s| s.as_str()) {
+            None if name == "main" => *annotation = Some(FunctionAnnotation::Entry),
+            Some("static") => *annotation = Some(FunctionAnnotation::Static),
+            _ => {}
+        }
+    }
+
+    for node in ast.inner.iter_mut() {
+        annotate_ast(node);
+    }
+}
+
+#[derive(Serialize)]
+pub struct ClangAst {
+    pub asts: HashMap<String, Node<Clang>>,
+    pub source_representation: Id,
+}
+
+impl std::fmt::Display for ClangAst {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "C ASTs for {:?}", self.asts.keys())
+    }
+}
+
+impl Representation for ClangAst {
+    fn name(&self) -> &'static str {
+        "clang_ast"
```

Comment:
Currently representation names are UpperCamelCase. I don't disagree with lower snake case (it is more consistent with tool names), but we should be consistent across all Representations.